publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/batch/dynamodb/LoadDynamodbResourceBatchJobHandler.java

```diff
@@ -0,0 +1,226 @@
+        }
+
+        logger.info("Processing batch with job type: {}, starting from marker: {}",
+                    input.getJobType(), input.getStartMarker());
+
+        var scanResult = dynamoDbClient.scan(getScanRequest(input.getStartMarker()));
+
+        var workItems = scanResult.getItems().stream()
+            .map(item -> {
+                var dynamoDbKey = new DynamodbResourceBatchDynamoDbKey(
+                    item.get(PRIMARY_KEY_PARTITION_KEY_NAME).getS(),
+                    item.get(PRIMARY_KEY_SORT_KEY_NAME).getS());
+                return new BatchWorkItem(dynamoDbKey, input.getJobType());
+            }).toList();
+
+        var batches = Lists.partition(workItems, SQS_BATCH_SIZE);
+        
+        var messagesQueued = CompletableFuture
+            .supplyAsync(() -> batches.parallelStream()
+                .mapToInt(this::sendBatchToQueue)
+                .sum(), forkJoinPool)
+            .exceptionally(e -> {
+                logger.error("Failed to process batches in parallel, falling back to sequential", e);
+                return batches.stream()
+                    .mapToInt(this::sendBatchToQueue)
+                    .sum();
+            })
+            .join();
+
+        logger.info("Processed {} items, queued {} messages", workItems.size(), messagesQueued);
+
+        var lastEvaluatedKey = scanResult.getLastEvaluatedKey();
+        if (lastEvaluatedKey != null && !lastEvaluatedKey.isEmpty()) {
+            sendEventForNextBatch(input, lastEvaluatedKey, context);
+        } else {
+            logger.info("Completed loading entire table for job type: {}", input.getJobType());
+        }
+
+        return new LoadDynamodbResponse(workItems.size(), messagesQueued, input.getJobType());
+    }
+
+    private void validateInput(LoadDynamodbRequest input) {
+        if (input == null || input.getJobType() == null || input.getJobType().isBlank()) {
+            throw new IllegalArgumentException(
+                "Missing required field 'jobType'. This lambda requires a jobType to be specified in the input. " +
+                "Example input: {\"jobType\": \"REINDEX_RECORD\"}"
+            );
+        }
+    }
+
+    private boolean isProcessingEnabled() {
+        var enabled = Optional.of(processingEnabled).orElse("true");
+        return "true".equalsIgnoreCase(enabled);
+    }
+
+    private ScanRequest getScanRequest(Map<String, AttributeValue> lastEvaluatedKey) {
+        var scanRequest = new ScanRequest()
+                                      .withTableName(tableName)
+                                      .withLimit(SCAN_PAGE_SIZE)
+                                      .withAttributesToGet(PRIMARY_KEY_PARTITION_KEY_NAME, PRIMARY_KEY_SORT_KEY_NAME);
+
+        if (lastEvaluatedKey != null && !lastEvaluatedKey.isEmpty()) {
+            scanRequest.withExclusiveStartKey(lastEvaluatedKey);
+        }
+
+        return scanRequest;
+    }
```

Comment:
BatchScan based on PK0 iterates through IdEntries and LogEntries as well, I think this batch will take about 2x the time of current batch scan (which is iterating through PK2)